modules/sdk-coin-algo/src/lib/transactionBuilder.ts

```diff
@@ -325,9 +327,18 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
 
   /** @inheritdoc */
   protected signImplementation({ key }: BaseKey): Transaction {
-    const buffKey = Utils.decodeSeed(key);
-    const keypair = new KeyPair({ prv: Buffer.from(buffKey.seed).toString('hex') });
-    this._keyPairs.push(keypair);
+    try {
+      const buffKey = Utils.decodeSeed(key);
```

Comment:
Nit: It's not clear from this line of code why this would throw if we pass a root key instead of an algo seed. Future readers of the code would have to dig into the decodeSeed function to understand that the root prv wouldnt pass the 36 byte length check of an algo key. A comment outlining why we could throw at this call would be useful.

Reply:
I will add a comment in a diff pr if you dont mind, i want to merge this to get it working in WP by friday

Reply:
yeah go for it